kedro/framework/session/session.py

```diff
@@ -31,14 +31,10 @@
 _active_session = None
 
 
-def _activate_session(session: "KedroSession", force: bool = False) -> None:
+def _activate_session(
+    session: "KedroSession", force: bool = False  # pylint: disable=unused-argument
+) -> None:
     global _active_session
-
-    if _active_session and not force and session is not _active_session:
-        raise RuntimeError(
-            "Cannot activate the session as another active session already exists."
-        )
-
     _active_session = session
```

Comment:
Given that https://github.com/kedro-org/kedro/issues/1458 is a prerequisite for this, will we need any of this stuff at all any more?

I'm a bit confused by whether the plan is to simultaneously do both of these:
1. introduce `after_context_created` hook
2. remove all the `_active_session` stuff

Or whether we want to do:
1. remove `_activate_session` stuff but leave `_active_session` global there
2. introduce `after_context_created` hook
3. remove all the `_active_session` stuff

i.e. do we do a straight swap in 0.18.1 (replace all `_active_session` things with the new hook) or do we do a phased introduction (in 0.18.1 you can use `_active_session` OR the new hook; then in 0.18.2 you can only use the new hook)?

Would be good to know if this affects @Galileo-Galilei 🙂

Reply:
I think the idea is to do 1&2 together in `0.18.1`.  Doing 1 only would leave the `_active_session` None only? Sorry I found that there are commits in my local repository but never pushed to the remote once as I was switching between these related branches.

Reply:
For what it's worth, I am completely ok with scenario (1), i.e. shipping the new hook and removing the session activation mechanism simultaneously in kedro==0.18.1. This is a private method and I think we can safely assume that people using my dirty hack of importing the global variable in their own plugin know what they are doing and accept the inherent risk of doing so. 

There is always a possibility that we miss something and that releasing everything at the same time will eventually break the plugin with no workaround at the plugin. I'll come back complaining in a new issue if it were to happen :). I have some concerns about the current proposal to make the context "frozen", but I'll let a comment in the other PR.